main/src/CoffeeMachineSimulator.java

```diff
@@ -0,0 +1,200 @@
+import java.util.Scanner;
+
+public class CoffeeMachineSimulator {
+
+    //declaring our class variables
+    int water = 400;
+    int milk = 540;
+    int coffeeBeans = 120;
+    int disCups = 9;
+    int money = 550;
+    String buy = "buy";
+    String fill = "fill";
+    String take = "take";
+    String exit = "exit";
+    String remaining = "remaining";
+    String back = "back";
+    int mode = 1;
+
+
+    // method myMethod check coffee machine mode, and call appropriate method
+    public void myMethod(String inputString) {
+
+        switch (mode) {
+            case 1:
+                action(inputString);
+                break;
+            case 2:
+                buy();
+                break;
+        }
+    }
+
+   // action method which check input, change machine mode and call appropriate method
+    public void action(String actionType) {
+
+        switch (actionType) {
+            case "buy":
+                mode = 2;
```

Comment:
you set mode to 2, and after that in buy function its become a 1, what is the meaning of this ?

Reply:
it was been in a homework task, that we must store current state, in any case i think it not needed